base/Basics.lua

```diff
@@ -4,15 +4,22 @@ require('CeresStdLib.base.Rawcode')
 require('CeresStdLib.base.Execute')
 require('CeresStdLib.base.Log')
 require('CeresStdLib.base.Init')
+require('CeresStdLib.base.Const')
 
 ANIMATION_PERIOD = 1 / 32.
 MAX_COLLISION_SIZE = 197.
 
-HEIGHT_ENABLER = fromRawCode('Amrf')
-TREE_RECOGNITION = fromRawCode('Aeat')
-LOCUST_ID = fromRawCode('Aloc')
-GHOST_INVIS_ID = fromRawCode('Agho')
-GHOST_VIS_ID = fromRawCode('Aeth')
+HEIGHT_ENABLER      = fromRawCode('Amrf')
+TREE_RECOGNITION    = fromRawCode('Aeat')
+LOCUST_ID           = fromRawCode('Aloc')
+GHOST_INVIS_ID      = fromRawCode('Agho')
+GHOST_VIS_ID        = fromRawCode('Aeth')
+DETECT_LEAVE        = fromRawCode('Amdf')
 
 DUMMY_PLAYER = players[PLAYER_NEUTRAL_PASSIVE]
 DUMMY_HOSTILE_PLAYER = players[PLAYER_NEUTRAL_AGGRESSIVE]
+
+local __oldInitB = InitBlizzard
```

Comment:
1. Can it ever fail?
2. Use replaceNative

Reply:
For some reason, in my copy of the game, that function appeared to be faulty, causing the thread to terminate before it can reach InitGlobals and so on.